src/index.ts

```diff
@@ -86,10 +96,28 @@ export async function gru<MasterArgs = undefined>(
 
     if (cluster.isWorker) {
         const masterArgsInWorker = await getMasterArgsInWorker()
-        const workerStart = options.start({
-            id: cluster.worker.id.toString(),
-            masterArgs: masterArgsInWorker,
-        })
+
+        let workerStart: null | void | Promise<any> = null
+
+        // Is this worker a dedicated worker?
+        // if dedicated, get the dedicated worker name then call the start function with that name
+        if (options.dedicatedWorkers) {
+            const workerName = getWorkerName(cluster.worker.id, workerMap)
```

Comment:
Ah I see how this works.

master forks, creates a map to look up process id -> worker name
worker starts, it doesn't know if it's a dedicated or a normal worker
gets args from master process, includes worker map
can now see that it's process id is a dedicated worker, starts that dedicated worker.

Another options I can think of, unsure if it's better

fork({ GRU_DEDICATED_WORKER_NAME: workerName, ...env })
then in this code process.env.GRU_DEDICATED_WORKER_NAME is set, and it can just call that?

Reply:
I tried this out and it does work. I'm not sure whether or not its better. Some data structure is still required to allow the master to check for the existence of all of the dedicated workers.

Reply:
fork({ GRU_DEDICATED_WORKER_NAME: workerName, ...env })

I've pushed a commit that does this. It avoids the need to get the workerMap into the worker processes.

Reply:
Nice!